streamlit/streamlit
# Refactor withMapboxToken to use HostConfig mapbox token

## Describe your changes
- remove commandLine as it looks like we're not using it
- remove MapboxToken.ts as  there doesn't seem to be a need for it
  - because this is removed, the implementation will go into `withMapboxToken.tsx`
- add LibContext into `withMapboxToken.tsx` in order to not make a request call if `mapboxToken` is defined
## GitHub Issue Link (if applicable)

## Testing Plan

- Explanation of why no additional tests are needed
- Unit Tests (JS and/or Python)
  - updated JS tests
- E2E Tests
- Any manual testing needed?
- I manually tested the command line changes by doing:
1. streamlit hello
2. change the mapboxToken to "test" within `.streamlit/config.toml`
3. run streamlit run e2e/scripts/st_pydeck_chart.py
4. check that the new mapboxToken was used and **THE RESULT WAS THE NEW TOKEN WAS USED!**

Another thing that I noticed was when we changed it to "test" was this weird behavior:
<img width="903" alt="Screenshot 2023-08-23 at 5 27 04 PM" src="https://github.com/streamlit/streamlit/assets/103006371/6bfdf123-ed95-45d9-9977-78fb3d099aee">

![Screenshot 2023-08-23 at 5 28 12 PM](https://github.com/streamlit/streamlit/assets/103006371/769e47a3-7b24-4459-9d9a-c3533737f65a)

From these screenshots, we can see some of the deckGL renders but the background doesn't fully render which makes sense because the mapbox token is bad. I think it's fine that we do a bad render as from what I looked through, i couldn't figure out how to catch the error above.

What it should look like:
<img width="883" alt="Screenshot 2023-08-23 at 5 28 54 PM" src="https://github.com/streamlit/streamlit/assets/103006371/f88e488b-760d-4363-8c47-37199fe0097e">

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.
